File: chapter4/java/designpattern/src/main/java/com/designpattern/ApplePayPaymentStrategy.java
package com.designpattern;

/**
 * ApplePayPaymentStrategy as ConcreteStrategy Classes.
 * ApplePayPaymentStrategy implementation of PaymentStrategy interface with an ApplePay payment algorithm.
 *
 * @Author Bridget Wu
 */
public class ApplePayPaymentStrategy implements PaymentStrategy {

    /**
     * ApplePay Payment implementation detail
     *
     * @param price
     */
    @Override
    public void pay(int price) {
        System.out.println("Charged total prices $" + price + " from ApplePay.");
    }
}

File: chapter4/typescript/ApplePayPaymentStrategy.ts
import { PaymentStrategy } from "./PaymentStrategy";

/**
 * ApplePayPaymentStrategy as ConcreteStrategy Classes.
 * ApplePayPaymentStrategy implementation of Strategy interface - PaymentStrategy implements an ApplePay payment algorithm.
 *
 * @Author Bridget Wu
 */
 export class ApplePayPaymentStrategy implements PaymentStrategy {
    /**
     * ApplePay Payment implementation detail
     *
     * @param price
     */
    public pay(price: number): void {
        console.log("Charged total prices $" + price + " from ApplePay.");
    }
}
